src/_P112_AS7265x.ino

```diff
@@ -171,13 +171,176 @@ boolean Plugin_112(byte function, struct EventStruct *event, String& string)
         static_cast<P112_data_struct *>(getPluginTaskData(event->TaskIndex));
 
       if (nullptr != P112_data) {
-        if (P112_data->sensor.dataAvailable()) {
-          // Measurement was succesfull, schedule a read.
-          P112_data->MeasurementStatus = 2;
-          Scheduler.schedule_task_device_timer(event->TaskIndex, millis() + 10);
-          P112_data->sensor.disableBulb(AS7265x_LED_WHITE);
-          P112_data->sensor.disableBulb(AS7265x_LED_IR);
-          P112_data->sensor.disableBulb(AS7265x_LED_UV);
+        if ((P112_data->sensor.dataAvailable()) or (P112_data->MeasurementStatus >= 2)) {
+
+          String RuleEvent;
+          RuleEvent  = getTaskDeviceName(event->TaskIndex);
+          RuleEvent += '#';
+
+          switch (P112_data->MeasurementStatus) {
+            case 1:
+              P112_data->sensor.disableBulb(AS7265x_LED_WHITE);
+              P112_data->sensor.disableBulb(AS7265x_LED_IR);
+              P112_data->sensor.disableBulb(AS7265x_LED_UV);
+
+              if (PCONFIG(3)) {
+                eventQueue.add(RuleEvent + "410=" + P112_data->sensor.getCalibratedA());

```

Comment:
Please use `F()` macro for strings in code, like this:
```c++
eventQueue.add(RuleEvent + F("410=") + P112_data->sensor.getCalibratedA());
```
If the String concat function does not allow it like this, you can also wrap it into `String(F("410="))`

Reply:
Corrected